packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -264,86 +333,171 @@ export function makeVirtualObjectManager(
-   * set, but eventually we'll use refcounts within virtual data to figure
-   * out when the vref becomes unreachable, allowing the vat to send a
+   * These were Presences at one point. We add to this set whenever we store a
+   * Presence into the state of a virtual object, or into a value in a
+   * makeVirtualScalarWeakMap() instance. We currently never remove anything
+   * from the set, but eventually we'll use refcounts within virtual data to
+   * figure out when the vref becomes unreachable, allowing the vat to send a
    * dropImport into the kernel and release the object.
-   *
    */
   /** @type {Set<string>} of vrefs */
-  const reachableVrefs = new Set();
+  const reachablePresences = new Set();
 
-  // We track imports, to preserve their vrefs against syscall.dropImport
-  // when the Presence goes away.
-  function addReachablePresenceRef(vref) {
-    // XXX TODO including virtual objects gives lie to the name, but this hack should go away with VO refcounts
-    const { type, allocatedByVat, virtual } = parseVatSlot(vref);
-    if (type === 'object' && (!allocatedByVat || virtual)) {
-      reachableVrefs.add(vref);
+  /**
+   * Set of all Remotables which are reachable by our virtualized data, e.g.
+   * `makeWeakStore().set(key, remotable)` or `virtualObject.state.foo =
+   * remotable`. The serialization process stores the Remotable's vref to disk,
+   * but doesn't actually retain the Remotable. To correctly unserialize that
+   * offline data later, we must ensure the Remotable remains alive. This Set
+   * keeps a strong reference to the Remotable. We currently never remove
+   * anything from the set, but eventually refcounts will let us discover when a
+   * Remotable no longer reachable via a virtual reference.  At that point we'll
+   * drop the strong reference from this set and thus permit garbage collection
+   * if the Remotable is no longer referenced in memory.
+   */
+  /** @type {Set<Object>} of Remotables */
+  const reachableRemotables = new Set();
+
+  // Note that since the `reachablePresences` set is keyed by vref,
+  // `processDeadSet` must query it directly in order to determine if a presence
+  // that found its way into the dead set is live or not, whereas it never needs
+  // to query the `reachableRemotables` set because that set holds actual live
+  // references and so Remotable references will only find their way into the
+  // dead set if they are actually unreferenced (including, notably, their
+  // absence from the `reachableRemotables` set, which will (eventually) be
+  // effectuated by reference counting).
+
+  function addReachableVref(vref) {
+    const { type, virtual, allocatedByVat } = parseVatSlot(vref);
+    if (type === 'object') {
+      if (allocatedByVat) {
+        if (virtual) {
+          incRefCount(vref);
+        } else {
+          // exported non-virtual object: Remotable
+          const remotable = getValForSlot(vref);
```

Comment:
RE the unmetered box comment above.. I suspect this sort of`getValForSlot` *doesn't* need to be unmetered, because it's a "this should never happen" error check (because of the `assert` on the next line). We shouldn't be able to get into that state.